### ISSUE TITLE
Suppression file not working with owasp/dependency-check-action

### ISSUE THREAD

**Issue by Yaytay** [dependency-check/DependencyCheck#6822]
**Describe the bug**
I have a java project called "token-validator", that, unfortunately I cannot share.
There is a file "dependency-check-supressions.xml" in the root of the project.
When I run this command (from the root of the project, after a successful build) it generates a false positive:
```
docker run -it -v .:/work --entrypoint "/bin/ash" owasp/dependency-check-action -c 'cd /work ; /usr/share/dependency-check/bin/dependency-check.sh --scan "./target" --format ALL --project "token-validator" --nvdApiKey "$NVD_API_KEY" --suppression dependency-check-supressions.xml  --failOnCVSS 7'
```
The false positive is:
```
token-validator-2.0.2-2-main-sources.jar: CVE-2021-3765(7.5)
```

I have tried experimenting with various suppression forms, but the interesting thing is that (with the three suppressions included in the gist) the output includes:
```
[INFO] Suppression Rule had zero matches: SuppressionRule{filePath=PropertyType{value=.*\/target\/token-validator-.*-main-javadoc\.jar, regex=true, caseSensitive=false},cpe={PropertyType{value=cpe:2.3:a:validator_project:validator:2.0.2.8:*:*:*:*:*:*:*, regex=false, caseSensitive=false},}}
[INFO] Suppression Rule had zero matches: SuppressionRule{sha1=89a7a3809632349c3ea207b290f9fb34999cc963,cve={CVE-2021-3765,}}
```
So it know that it mismatched on two and presumably that means it did match on the other one (filePath and cve, the one I'd most like to work) but it still reports:
```
[ERROR]

One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '7.0':

token-validator-2.0.2-2-main-sources.jar: CVE-2021-3765(7.5)
```

**Version of dependency-check used**
10.0.2

**Log file**
https://gist.github.com/Yaytay/d648037357ba765af935197d9cf6463e
This gist includes all of the output from dependency-check and the suppressions file.

**To Reproduce**
Create Java (maven) project called token-validator, build it to the target directory (I suspect it doesn't matter what the project does as a long as the version number is below v13.

**Expected behavior**
If the suppression is not reported by the "Unused Suppression Rule Analyzer" I expect any alerts that match it to be suppressed.

**Additional information**
After running, the contents of the root of my project are:
```
-rwxr--r-- 1 yaytay  yaytay     267 Nov 20  2023 checkstyle-suppressions.xml
-rwxr--r-- 1 yaytay  yaytay    3460 Nov 20  2023 checkstyle.xml
-rw-r--r-- 1 root    root      2868 Jul  9 08:17 dependency-check-gitlab.json
-rw-r--r-- 1 root    root     30563 Jul  9 08:17 dependency-check-jenkins.html
-rw-r--r-- 1 root    root      9202 Jul  9 08:17 dependency-check-junit.xml
-rw-r--r-- 1 root    root       939 Jul  9 08:17 dependency-check-report.csv
-rw-r--r-- 1 root    root    246602 Jul  9 08:17 dependency-check-report.html
-rw-r--r-- 1 root    root     79628 Jul  9 08:17 dependency-check-report.json
-rw-r--r-- 1 root    root    239872 Jul  9 08:17 dependency-check-report.log
-rw-r--r-- 1 root    root     17264 Jul  9 08:17 dependency-check-report.sarif
-rw-r--r-- 1 root    root     91463 Jul  9 08:17 dependency-check-report.xml
-rwxr--r-- 1 yaytay  yaytay     778 Jul  9 08:04 dependency-check-supressions.xml
-rwxr--r-- 1 yaytay  yaytay      74 Nov 20  2023 findbugsFilterFile.xml
-rwxr--r-- 1 yaytay  yaytay      74 Nov 20  2023 findbugsSecurityFilterFile.xml
-rwxr--r-- 1 yaytay  yaytay    1695 Nov 15  2023 gitlab_settings.xml
-rwxr--r-- 1 yaytay  yaytay    6743 Jul  6 14:47 nbactions.xml
drwxr-xr-x 1 yaytay  yaytay     512 Nov 21  2023 nbproject
-rwxr--r-- 1 yaytay  yaytay   19076 Jul  7 20:26 pom.xml
-rwxr--r-- 1 yaytay  yaytay    7902 Jan  4  2024 README.md
drwxr-xr-x 1 yaytay  yaytay     512 Nov 15  2023 src
drwxr-xr-x 1 yaytay  yaytay     512 Jul  7 20:26 target
```

**Comment by Yaytay** [comment]
Incidentally, the suppression rule with the sha is the one generated from the report HTML file, that doesn't match.
I'm not rebuilding the project between runs, just rerunning the one docker command, so the sha is not changing.

**Comment by chadlwilson** [comment]
If your build recompiles the project every time before the scan, it’s reasonable to assume the hash is different every time, so a Sha based suppression is not the right approach. It sounds like you’d be better to use the maven plugin or otherwise use excludes to tell it to not look at various directories which include your own project’s build output and instead focus on its dependencies as understood from Maven?

**Comment by Yaytay** [comment]
I'm sorry @chadlwilson but I have clearly not articulated the point correctly.

I am not bothered by the two suppressions that failed to match (though I'd be interesting in knowing why they did), the bug is that the other suppression presumably did match and the build still errored.


I'd be happy to discuss other points you raise but I fear that doing so would detract from that key point - the suppression appears to be matching but not suppressing.

**Comment by chadlwilson** [comment]
Well it didn’t match on the file path + CVE one either because the file path does not match. (javadoc != sources)

So really it’s just a question of why all 3 didn’t show up as unused suppressions, no?

**Comment by Yaytay** [comment]
Oh stupid me!
There are a few jars in that dir and I bet it's been bouncing back and forth between which one was failing each time I updated the suppression and I didn't even notice that the suffx was different.

It has now successfully passed after I made the regex accept anything in that position.

And that also explains why it wasn't reported as not-matching - it did match a file, just not all files.

Thank you for providing the eyes/brain that I was missing.
